build/linter/unconvert/analysis.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package unconvert
+
+import (
+	"fmt"
+	"go/token"
+	"go/types"
+
+	unconvertAPI "github.com/golangci/unconvert"
+	"github.com/pingcap/tidb/build/linter/util"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/loader"
+)
+
+// Name is the name of the analyzer.
+const Name = "unconvert"
+
+// Analyzer is the analyzer struct of unconvert.
+var Analyzer = &analysis.Analyzer{
+	Name: Name,
+	Doc:  "Remove unnecessary type conversions",
+	Run:  run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
```

Comment:
Later you should add a method document or file document about all these analyzers' semantic. For example, what case is good and what case is bad. It's _not_ intuitive for developer reading the code and understand the effective rule.